### PR TITLE
#12075 :- Customer Email with diacritics isn't allowed

### LINKED ISSUE #12075: Customer Email with diacritics isn't allowed
Magento2 doesn't allow E-Mails with diacritics in CURL Requests.
According to [wikipedia](https://en.wikipedia.org/wiki/Email_address#Internationalization) this should be allowed, but Magento is restricting it. 

Is there a Fix or Workaround or something?

### Preconditions
1. Magento Version 2.1.8

### Steps to reproduce
1. get email address with diacritics (pepé@emailtest.com)
2. create json data object with email 
`$data = [
                    'customer' =>
                    [
                        'email' => pepé@emailtest.com
                     ]
]`
3. create user using curl via /index.php/rest/V1/customers

### Expected result
1. User should be created with correct email address

### Actual result
1.` {"message":"\"Email\" ist keine g\u00fcltige E-Mail-Adresse."` (roughly translated: Email isn't a valid E-Mail-Address)

### Additional Notes
Reproduced it via Swagger with using the following query  /swagger#/customerAccountManagementV1/customerAccountManagementV1CreateAccountPost
swagger_create customer
The response is : 
```
{
  "message": "\"Email\" is not a valid email address."
}
```
But this is also reproducible if try to create a customer from Magento Admin
email_from admin

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will fix the issue with the email address validation for email address internationalization (EAI) 

### Fixed Issues (if relevant)

1. Fixes magento/magento2#12075

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by konarshankar07** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/Validator/EmailAddress.php [hunk 1]
```diff
@@ -6,36 +6,9 @@
  * See COPYING.txt for license details.
  */
 namespace Magento\Framework\Validator;
+use Laminas\Validator\EmailAddress as LaminasEmailValidator;
 
-class EmailAddress extends \Zend_Validate_EmailAddress implements \Magento\Framework\Validator\ValidatorInterface
+class EmailAddress extends LaminasEmailValidator
```
Comment (rogyar) [inline lib/internal/Magento/Framework/Validator/EmailAddress.php:11]: Hi @konarshankar07. Thank you for your work. Could you explain your fix, please? I see that you have removed the part that disables TLD validation. But validation of the name in the email (before @) still in place. 

Also, you've removed the interface mention from the validator. Could you shed some light on the reason, please? 

Thanks.

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
Hi @konarshankar07. Could you please look at https://github.com/magento/magento2/pull/31003#discussion_r528296362?
Thank you.

**Comment by engcom-Charlie** [comment]
Hi @konarshankar07, do you have any updates?

File: lib/internal/Magento/Framework/Validator/EmailAddress.php (hunk 1 above)
Comment (konarshankar07) [reply lib/internal/Magento/Framework/Validator/EmailAddress.php:11]: Hello @rogyar 
I think we should remove the zendframework1 dependency from the magento and use the Laminas dependency to the magento core that's why I removed the interface and TLD validation and extended the laminas Email validator.Let me know your thoughts
Thanks

Comment (rogyar) [reply lib/internal/Magento/Framework/Validator/EmailAddress.php:11]: Hi @konarshankar07. That makes sense, thank you for the explanation. 

In the current version, we have the TLD validation set to `false`. However, in Laminas hostname validator, the `useTldCheck` set to `true` by default. So it means that after your change the TLD will be validated (but should not). 

Also, still not obvious how this change will fix validation of the `THIS@mail.com` part of the email address that may contain diacritics. 

Could you clarify the mentioned points, please? 

Thanks.

**Comment by engcom-Charlie** [comment]
Hi @konarshankar07. Could you please take a look at https://github.com/magento/magento2/pull/31003#discussion_r535095984?
Thank you.

**Comment by engcom-Charlie** [comment]
@konarshankar07 @rogyar with changes from this PR we have the next error during install:
`PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Magento\Framework\Validator\DataObject::addRule() must implement interface Zend_Validate_Interface, instance of Magento\Framework\Validator\EmailAddress given`
At first, needs to change \Magento\Framework\Validator\DataObject::addRule parameter type which will affect other validators. You can fix it but this change could be delivered only to 2.5 version since \Magento\Framework\Validator\DataObject marked as api.

**Comment by engcom-Charlie** [comment]
@konarshankar07 please fix the issue mentioned above, I created a separate issue for removing a dependency here https://github.com/magento/magento2/issues/31511.
Thank you.

**Comment by konarshankar07** [comment]
Hello @engcom-Charlie 
Thanks for the update and I will fix this issue using the zendframework1

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run all tests

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @konarshankar07, can you update this PR with the latest mainline also have a look at the failed tests?

Thanks!

**Comment by engcom-Foxtrot** [comment]
Hello, @konarshankar07. Unfortunately, we are closing this PR due to inactivity. If you want to continue your work on this issue, please one new one. Also, please do not replace the Zend_Validate with Laminas as such change poses a high risk.

**Pull request closed without merging**